### PR TITLE
Update gemrb to 0.9.3

### PR DESCRIPTION
**Pull request by Peppersawce**
Working recipe for building gemrb 0.9.3 on Beta 5 + files needed for it. Build process works but could be better, there's an issue with cmake finding extend2da.py at buildtime that's being fixed in an unorthodox way. Improvements are welcome.

Discussion here: https://discuss.haiku-os.org/t/gemrb-0-9-3-port-questions-about-haikuports-recipes/15535

### CODE REVIEW

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (Begasus) [inline games-engines/gemrb/gemrb-0.9.3.recipe:23]: Why mark as broken, 32bit hasn't been tested yet :)

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (Begasus) [inline games-engines/gemrb/gemrb-0.9.3.recipe:75]: Empty line between PATCH() and BUILD()

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (Begasus) [inline games-engines/gemrb/gemrb-0.9.3.recipe:77]: for 32bit this probably needs secondaryArchSubDir

File: games-engines/gemrb/additional-files/cmake_install.cmake
Comment (Begasus) [inline games-engines/gemrb/additional-files/cmake_install.cmake:84]: Would be nice if the DESTINATION paths would use a variable instead of a hardcoded path

**Review by Begasus — CHANGES_REQUESTED** [review]
Needs some work, thanks so far!

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (jmairboeck) [reply games-engines/gemrb/gemrb-0.9.3.recipe:75]: Actually, `/usr/bin` doesn't make sense on Haiku. Is this for the shebang line? In that case, the path could be removed completely, so just `python3` should work. Otherwise, `/bin/python3` is more correct.

File: games-engines/gemrb/additional-files/gemrb.rdef
Comment (jmairboeck) [inline games-engines/gemrb/additional-files/gemrb.rdef:4]: Usually, these are set as "variables" which are replaced with sed in the INSTALL function with values derived from `$portVersion`, and therefore this file has a `.rdef.in` file extension. This is so that it doesn't need to be updated manually for every new version.

You'll find many examples in HaikuPorts how to do this.

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (jmairboeck) [inline games-engines/gemrb/gemrb-0.9.3.recipe:93]: Why not use `$docDir`? That includes the package name in the path.

**Comment by Peppersawce** [comment]
I reworked the recipe with @Begasus 's feedback and managed to find a solution with sed to fix the build issue cmake has (something about an extra "/")
I'll look around poking with @jmairboeck later as well :)

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (Peppersawce) [reply games-engines/gemrb/gemrb-0.9.3.recipe:76]: Sounds like a good idea but I have no way to test it on a 32bit system atm and I don't feel like making changes I can't test

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (Peppersawce) [reply games-engines/gemrb/gemrb-0.9.3.recipe:74]: Yeah, it's for the shebang line

File: games-engines/gemrb/additional-files/GemRB.cfg
Comment (Peppersawce) [reply games-engines/gemrb/additional-files/GemRB.cfg:245]: About that, currently  $settingsDir is the one under system for some reason, which is a less-than-ideal solution as well. I'm honestly slightly confused about where things should go, what would be the proper path(s) where I should store the .cfg and saves in userspace?

**Comment by Peppersawce** [comment]
Removed the pre-made cfg file and added sed commands to edit the source's one instead. I've tried editing the config to use better paths but the executable only seems to be able to write in "Home" and I have no clue why tbh

File: games-engines/gemrb/additional-files/GemRB.cfg
Comment (Peppersawce) [reply games-engines/gemrb/additional-files/GemRB.cfg]: I agree, but at the moment it seems like the program only has write access to Home, any way to fix that?

File: games-engines/gemrb/additional-files/GemRB.cfg
Comment (Peppersawce) [reply games-engines/gemrb/additional-files/GemRB.cfg]: Edited path to "finddir B_USER_SETTINGS_DIRECTORY + /gemrb" as suggested, there's an issue with this path however, the program will not save games if the ~/config/settings/gemrb folder doesn't already exist

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
Thanks so far, I'll let the others comment if anything else should need some ironing.

**Comment by jmairboeck** [comment]
> Mmm...
> 
> In `/system/data/cmake/Modules/FindPython/Support.cmake` I see way too many lines that look lie this:
> 
> ```cmake
> list (TRANSFORM suffixes PREPEND "lib/python${version}/config-${version}")
> ```
> 
> And none with "lib/x86/python${version}/config-${version}", so that might explain why it fails on 32 bits.
> 
> So... our cmake is pretty broken right now (in regard to finding Python, at least :-D).
> 
> Do we have someone that understand cmake well enough to fix that?

There should be a CMake variable `${CMAKE_HAIKU_SECONDARY_ARCH_SUBDIR}` (see [here](https://github.com/Kitware/CMake/blob/ac8738c0770667086d43aba024dc749602252011/Modules/Platform/Haiku.cmake#L43)) which could be used here (appended after `lib`).

File: games-engines/gemrb/patches/gemrb-0.9.3.patchset
Comment (Begasus) [reply games-engines/gemrb/patches/gemrb-0.9.3.patchset:2]: I think this is done by "hp -e" extracting the patchset, this is set in haikuports.conf file in the line PACKAGER=?

Comment (jmairboeck) [reply games-engines/gemrb/patches/gemrb-0.9.3.patchset:2]: No, haikuporter uses whatever user information the commits in the work repo have. These aren't created by haikuporter but with regular `git commit`. The PACKAGER in haikuports.conf is just for the metadata of the created packages.

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (Peppersawce) [reply games-engines/gemrb/gemrb-0.9.3.recipe:100]: If that code just makes the directories it's not enough for the cache issue, there's something more to it unfortunately.

It would solve the saves issue though so I will add that to the recipe :)

File: games-engines/gemrb/patches/gemrb-0.9.3.patchset
Comment (Peppersawce) [reply games-engines/gemrb/patches/gemrb-0.9.3.patchset:2]: If it's fine with you all and there's no guideline or anything, I'd rather keep it this way, I get a chuckle out of that kind of headers ngl

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (Peppersawce) [reply games-engines/gemrb/gemrb-0.9.3.recipe:100]: Nah I get it, having the cache in home is not ideal and I'd rather have that sorted as well

File: games-engines/gemrb/gemrb-0.9.3.recipe
Comment (Peppersawce) [reply games-engines/gemrb/gemrb-0.9.3.recipe:35]: Let me get this straight: if the recipe doesn't build on gcc2 then $secondaryArchSuffix only goes on the "provides" with the package name and not on cmd: or app: ?

**Pull request merged**
